Durcir les guardrails anti-éparpillement (hook + auditor)

## Contexte

Issue #3676 (epic). Volet **outillage** (2ᵉ partie du ticket) : durcir les guardrails Claude pour empêcher la **réintroduction future** de règles métier hors `~/modules/domain`. Indépendant du code applicatif → **aucune dépendance** (tourne en parallèle de T1). Décision actée : *hook élargi (haute confiance) + structural-auditor ERROR sur les seuils*.

## Fichiers impactés

- `.claude/hooks/block-bad-patterns.sh` (modification)
- `.claude/agents/structural-auditor/AGENT.md` (modification, §2.15 Domain layer)
- `.claude/rules/automation.md` (modification — tableau des patterns bloqués + règle inline domaine)

## Changement attendu

**`block-bad-patterns.sh`** — élargir la détection SIREN/date hors `domain/` (haute confiance, faux positifs improbables) :
- Étendre la règle SIREN existante (`slice(0, 9)`) à : `slice(0,9)` (sans espace), `substring(0, 9)` / `substring(0,9)`, `substr(0, 9)` / `substr(0,9)`. Message : « Utilise `extractSiren()` de `~/modules/domain`. »
- Ajouter une règle d'**arithmétique de date inline** hors `domain/` : bloquer `\.getMonth\(\)` et `\.getDate\(\)` employés dans un calcul (au-delà du `getFullYear()` déjà bloqué). Message : pointer vers les helpers `campaign` de `~/modules/domain`. Conserver les mêmes exclusions que les règles voisines (`domain/`, `__tests__`, `.test.`, `.spec.`).
- **Ne PAS** bloquer les seuils 5/50/100 au niveau du hook (trop de faux positifs sur littéraux légitimes — c'est le rôle du structural-auditor, ci-dessous).

**`structural-auditor/AGENT.md` §2.15** :
- Passer « Hardcoded regulatory thresholds (5%, 50, 100) » de **[WARN] → [ERROR]**.
- Ajouter un check **anti-réimplémentation** : grep des helpers `domain` exportés mais réécrits inline hors `domain/` — au minimum `isCseRequired`, `isComplianceProcessRequired`, `isComplianceProcessRevisionRequired`, `isCancelled` (signaler une condition inline équivalente comme `cancelledAt !== null`, `workforce >= 100 && hasCse`). → **[ERROR]**.
- Ajouter un check **formatters dupliqués** : `toLocaleString("fr-FR")` hors `domain/` (doublon probable de `formatCount`/`formatRate`/`formatCurrency`). → **[WARN]**.

**`automation.md`** — mettre à jour le tableau « PreToolUse — block-bad-patterns.sh » (nouvelles entrées SIREN variantes + date) et renforcer la règle inline « toute règle métier → `~/modules/domain` ».

## Scénarios de test

Couvre S6, S7 :
- **Étant donné** le hook, **quand** un edit contient `siret.substring(0, 9)` dans un `.ts` hors `domain/`, **alors** l'edit est rejeté (exit 2).
- **Étant donné** le hook, **quand** un edit contient `slice(0, 9)` dans un fichier sous `domain/`, **alors** il passe (exclusion conservée).
- **Étant donné** le structural-auditor, **quand** un fichier hors `domain/` contient `>= 100` sur un effectif, **alors** il est remonté en **[ERROR]** (et non [WARN]).
- **Étant donné** le structural-auditor, **quand** un fichier réécrit `cancelledAt !== null`, **alors** un [ERROR] anti-réimplémentation est levé.

## Référence Figma

N/A.

## Critères d'acceptation

- [ ] Hook : variantes SIREN (`slice`/`substring`/`substr` `(0,9)` avec ou sans espace) bloquées hors `domain/` ; arithmétique de date (`getMonth`/`getDate`) bloquée hors `domain/` ; exclusions `domain/`/tests préservées.
- [ ] Hook : aucun blocage sur les littéraux 5/50/100 (laissé au structural-auditor).
- [ ] structural-auditor §2.15 : seuils en **[ERROR]** ; check anti-réimplémentation des helpers `domain` ; check `toLocaleString("fr-FR")` hors domaine.
- [ ] `automation.md` à jour (tableau patterns + règle inline domaine).
- [ ] Vérif manuelle : le hook ne casse pas un edit légitime du dépôt (faire tourner `block-bad-patterns.sh` sur un échantillon de fichiers récents non concernés).

## Complexité

**Taille : M (3 pts)** — outillage `.claude` uniquement (pas de code applicatif), mais rédaction soignée des regex pour éviter les faux positifs. Pas de label `complexe`.

--- a/.claude/agents/structural-auditor/AGENT.md
+++ b/.claude/agents/structural-auditor/AGENT.md
@@ -152,19 +152,31 @@ For changed files in `src/server/`:
 
 ```bash
 # Must return ZERO — no inline getFullYear outside domain/
-grep -rn "new Date()\.getFullYear()" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/" | grep -v "__tests__" | grep -v "\.test\."
+grep -rn "getFullYear()" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/" | grep -v "__tests__" | grep -v "\.test\."
 
-# Must return ZERO — no inline slice(0, 9) outside domain/
-grep -rn "slice(0, 9)" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/"
+# Must return ZERO — no inline slice/substring/substr(0, 9) outside domain/
+grep -rn "slice(0, *9)\|substring(0, *9)\|substr(0, *9)" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/" | grep -v "__tests__" | grep -v "\.test\."
+
+# Must return ZERO — no inline date arithmetic outside domain/
+grep -rn "\.getMonth()\|\.getDate()" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/" | grep -v "__tests__" | grep -v "\.test\."
 
 # Must return ZERO — no local function definitions duplicating domain
 grep -rn "function getCurrentYear\|function getCseYear\|function getSiren" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/"
+
+# Must return ZERO — no domain helpers reimplemented inline outside domain/
+grep -rn "cancelledAt !== null\|cancelledAt != null\|workforce >= 100\|effectifs >= 100\|workforce < 50\|effectifs < 50" src/ --include="*.ts" --include="*.tsx" | grep -v "domain/" | grep -v "__tests__" | grep -v "\.test\."
+
+# Must return ZERO (or WARN) — toLocaleString("fr-FR") outside domain/ (likely duplicates formatCount/formatRate/formatCurrency)
+grep -rn 'toLocaleString.*fr-FR\|toLocaleString.*"fr"' src/ --include="*.ts" --include="*.tsx" | grep -v "domain/" | grep -v "__tests__" | grep -v "\.test\."
 ```
 
-- Inline `new Date().getFullYear()` → **[ERROR]**
-- Inline `siret.slice(0, 9)` → **[ERROR]**
+- Inline `getFullYear()` → **[ERROR]**
+- Inline `slice/substring/substr(0, 9)` for SIREN extraction → **[ERROR]**
+- Inline `.getMonth()` / `.getDate()` for date calculations → **[ERROR]**
 - Local function duplicating domain → **[ERROR]**
-- Hardcoded regulatory thresholds (5%, 50, 100) → **[WARN]**
+- Hardcoded regulatory thresholds (5%, 50, 100) → **[ERROR]**
+- Inline condition reimplementing a domain helper (e.g. `cancelledAt !== null` instead of `isCancelled()`, `workforce >= 100 && hasCse` instead of `isCseRequired()`, `isComplianceProcessRequired()`, `isComplianceProcessRevisionRequired()`) → **[ERROR]**
+- `toLocaleString("fr-FR")` outside `domain/` — probable duplicate of `formatCount`/`formatRate`/`formatCurrency` → **[WARN]**
 
 #### 2.16 Accessibility (quick check)
 

--- a/.claude/hooks/block-bad-patterns.sh
+++ b/.claude/hooks/block-bad-patterns.sh
@@ -108,10 +108,21 @@ check_pattern '\.(ts|tsx)$' \
   'Inline getFullYear() is forbidden. Use getCurrentYear() or getCseYear() from ~/modules/domain.' \
   '(domain/|__tests__|\.test\.|\.spec\.)'
 
-# Domain layer — slice(0, 9) for SIREN extraction must come from ~/modules/domain
+# Domain layer — slice/substring/substr(0, 9) for SIREN extraction must come from ~/modules/domain
 check_pattern '\.(ts|tsx)$' \
   'slice\(0,[[:space:]]*9\)' \
   'Inline slice(0, 9) is forbidden. Use extractSiren() from ~/modules/domain.' \
+  '(domain/|__tests__|\.test\.|\.spec\.)'
+
+check_pattern '\.(ts|tsx)$' \
+  '(substring|substr)\(0,[[:space:]]*9\)' \
+  'Inline substring/substr(0, 9) is forbidden. Use extractSiren() from ~/modules/domain.' \
+  '(domain/|__tests__|\.test\.|\.spec\.)'
+
+# Domain layer — inline date arithmetic (getMonth/getDate) must use domain campaign helpers
+check_pattern '\.(ts|tsx)$' \
+  '\.(getMonth|getDate)\(\)' \
+  'Inline .getMonth()/.getDate() is forbidden. Use campaign date helpers from ~/modules/domain.' \
   '(domain/|__tests__|\.test\.|\.spec\.)'
 
 # Zod imports forbidden in router files — schemas must be in ~/modules/{domain}/schemas.ts

--- a/.claude/rules/automation.md
+++ b/.claude/rules/automation.md
@@ -34,7 +34,9 @@ Blocks edits containing forbidden patterns:
 | `dangerouslySetInnerHTML` | `.tsx/.jsx` | XSS risk — use safe rendering or DOMPurify |
 | `: any`, `as any` | `.ts/.tsx` (excl. test files) | Use `unknown` with type narrowing |
 | `getFullYear()` | `.ts/.tsx` (excl. `domain/`, tests) | Use `getCurrentYear()` / `getCseYear()` from `~/modules/domain` |
-| `slice(0, 9)` | `.ts/.tsx` (excl. `domain/`, tests) | Use `extractSiren()` from `~/modules/domain` |
+| `slice(0, 9)` / `slice(0,9)` | `.ts/.tsx` (excl. `domain/`, tests) | Use `extractSiren()` from `~/modules/domain` |
+| `substring(0, 9)` / `substring(0,9)` / `substr(0, 9)` / `substr(0,9)` | `.ts/.tsx` (excl. `domain/`, tests) | Use `extractSiren()` from `~/modules/domain` |
+| `.getMonth()` / `.getDate()` | `.ts/.tsx` (excl. `domain/`, tests) | Use campaign date helpers from `~/modules/domain` |
 | `from "zod"` | `routers/*.ts`, `.tsx` (excl. tests) | Import schemas from `~/modules/{domain}/schemas.ts` |
 | `#hex`, `rgb()`, `rgba()` | `.scss` | Use DSFR CSS custom properties |
 | Non-route `.tsx` in `src/app/` | `src/app/**/*.tsx` | Move to `src/modules/` and import from barrel |
@@ -105,9 +107,11 @@ Apply these rules **as you write code**, before any agent runs:
 **Domain layer:**
 
 - No `new Date().getFullYear()` → use `getCurrentYear()` or `getCseYear()` from `~/modules/domain`
-- No `siret.slice(0, 9)` → use `extractSiren(siret)` from `~/modules/domain`
+- No `siret.slice(0, 9)`, `substring(0, 9)`, or `substr(0, 9)` → use `extractSiren(siret)` from `~/modules/domain`
+- No `.getMonth()` / `.getDate()` in calculations → use campaign date helpers from `~/modules/domain`
 - No hardcoded thresholds (5%, 50, 100) → use named constants from `~/modules/domain`
 - No local `getCurrentYear`/`getCseYear`/`getSiren` function definitions → import from `~/modules/domain`
+- No inline reimplementation of domain helpers: `cancelledAt !== null` → `isCancelled()`, `workforce >= 100` → `isCseRequired()` / `isComplianceProcessRequired()` / `isComplianceProcessRevisionRequired()`
 - New business rules → add to `~/modules/domain` as pure functions with tests
 
 **Security:**

